moh_ingest.py

```diff
@@ -0,0 +1,248 @@
+
+        print(f"Loading {file_name}...")
+        payload = read_json(data_location + file_name)
+        if payload is not None:
+            if headers == "GET_AUTH_HEADER":
+                headers = auth.get_auth_header()
+            response = requests.post(
+                ingest_url, headers=headers, data=json.dumps(payload)
+            )
+
+            if response.status_code == HTTPStatus.CREATED:
+                print(f"INGEST OK 201! \nRETURN MESSAGE: {response.text}\n")
+            elif response.status_code == HTTPStatus.NOT_FOUND:
+                print(f"ERROR 404: {ingest_url} was not found! Please check the URL.")
+                break
+            else:
+                print(
+                    f"\nREQUEST STATUS CODE: {response.status_code} \nRETURN MESSAGE: {response.text}\n"
+                )
+                break
+    else:
+        ingest_finished = True
+
+    if ingest_finished:
+        print("All files have been processed.")
+    else:
+        print("Aborting processing due to an error.")
+
+
+def run_check(katsu_server_url, env_str, data_location, headers, ingest_version):
+    """
+    Run a series of checks to ensure that the ingest is ready to run.
+        - Check if the environment file exists
+        - Check if the environment variable is set
+        - Check if the Katsu server is running the correct version
+        - Check header authentication
+    """
+    # Check if environment file exists
+    if os.path.exists(env_str):
+        print("PASS: The environment file exists.")
+    else:
+        print("ERROR ENV CHECK: The environment file does not exist.")
+
+    # Check if environment variable is set
+    if data_location:
+        print("PASS: Data location is set.")
+    else:
+        print("ERROR LOCATION CHECK: data location is not set.")
+
+    # check if Katsu server is running correct version
+    version_check_url = katsu_server_url + "/api/v1/version_check"
+    try:
+        response = requests.get(version_check_url)
+        if response.status_code == HTTPStatus.OK:
+            katsu_version = response.json()["version"]
+            if check_api_version(
+                ingest_version=ingest_version, katsu_version=katsu_version
+            ):
+                print(f"PASS: Katsu server is running on a compatible version.")
+            else:
+                print(
+                    f"ERROR: Katsu server is running on {katsu_version}. Required version {ingest_version} or greater."
+                )
+        else:
+            print(f"ERROR VERSION CHECK {response.status_code}: {response.text}")
+    except ConnectionError as e:
+        print(f"ERROR VERSION CHECK: {e}")
+        return
+
+    # check authorization
+    if headers == "GET_AUTH_HEADER":
+        try:
+            headers = auth.get_auth_header()
+            print("PASS: Auth header is set.")
+        except Exception as e:
+            print(f"ERROR AUTH CHECK: {e}")
```

Comment:
On my setup, the Katsu version check won't work unless you also give the proper headers. Recommend swapping the order of the steps to the following, and adding headers=headers as an argument to the version_check_url get:
```suggestion
    # check authorization
    if headers == "GET_AUTH_HEADER":
        try:
            headers = auth.get_auth_header()
            print("PASS: Auth header is set.")
        except Exception as e:
            print(f"ERROR AUTH CHECK: {e}")

    # check if Katsu server is running correct version
    version_check_url = katsu_server_url + "/api/v1/version_check"
    try:
        response = requests.get(version_check_url, headers=headers)
        if response.status_code == HTTPStatus.OK:
            katsu_version = response.json()["version"]
            if check_api_version(
                ingest_version=ingest_version, katsu_version=katsu_version
            ):
                print(f"PASS: Katsu server is running on a compatible version.")
            else:
                print(
                    f"ERROR: Katsu server is running on {katsu_version}. Required version {ingest_version} or greater."
                )
        else:
            print(f"ERROR VERSION CHECK {response.status_code}: {response.text}")
    except ConnectionError as e:
        print(f"ERROR VERSION CHECK: {e}")
        return
```